btx/diagnostics/run.py

```diff
@@ -60,19 +73,26 @@ def finalize_powders(self):
         max, avg, and std dev (and gain mode counts if applicable).
         """
         self.powders_final = dict()
-        powder_max = np.array(self.comm.gather(self.powders['max'], root=0))
-        powder_min = np.array(self.comm.gather(self.powders['min'], root=0))
-        powder_sum = np.array(self.comm.gather(self.powders['sum'], root=0))
-        powder_sqr = np.array(self.comm.gather(self.powders['sqr'], root=0))
         total_n_proc = self.comm.reduce(self.n_proc, MPI.SUM)
+        if total_n_proc == 0:
+            powder_max = np.zeros(self.psi.det.shape())
+            powder_min = np.zeros(self.psi.det.shape())
+            powder_sum = np.zeros(self.psi.det.shape())
+            powder_sqr = np.zeros(self.psi.det.shape())
+        else:
```

Comment:
Please describe why this is needed?

Reply:
An occurs when this code is removed.

Reply:
Thanks Deeban, I see why it was needed now - per our discussion today, could you please revert it back to include your changes and I'll approve the PR. Thanks!